### PR TITLE
Ensure the transform identities are unique per execution history store

### PR DESCRIPTION
**Pull request by wolfs**
TL;DR: The uniqueId for project artifact transforms that is used in the execution history store is not unique, we fix it by adding the `buildTreePath` of the producer project to it.

The following is only about workspaces and transforms on project artifacts.

The unique workspace id of an artifact transform is used in two places:
* key in the execution history in the current build (Gradle scoped)
* workspace location in the producer project

That means the unique id needs to be unique:
* per artifact to transform in the producer project <- This is simple, we are currently using the relative path of the artifact to the root in the producer project
* per artifact to transform in the consumer build <- This is violated currently; see the two tests added in this PR.

To fix the problem, we can add additional data to the uniqueId related to the producer and consumer projects. Anything we'll add that is related to the consumer project will not help with the unique id problem, since the execution history is for the consumer build. So we need to add something from the producer project.

As the tests show, we must add the identity path to make it unique. That has the downside that if add the same build as an included build, then project artifacts within that included build will be re-executed since a new workspace will be assigned to them. Though while running the same included build constellation, everything will be re-used as expected.

As an additional advantage, this change makes the uniqueId unique within the whole build tree.

### CODE REVIEW

**Review by asodja — APPROVED** [review]
LGTM.

Makes sense to me. Are there any other performance impacts other then that artifact transform will rerun if workspace is changed?

**Comment by wolfs** [comment]
@bot-gradle test and merge
